llpc/util/llpcElfWriter.cpp

```diff
@@ -735,6 +737,285 @@ template <class Elf> void ElfWriter<Elf>::updateMetaNote(Context *pContext, cons
   pNewNote->data = data;
 }
 
+// =====================================================================================================================
+// Get the count of relocs in the reloc section
+//
+// @param pRelocSection : Reloc section
+template <class Elf> size_t ElfWriter<Elf>::getRelocsCount(const SectionBuffer *pRelocSection) {
+  size_t relocCount = 0;
+  if (pRelocSection->secHead.sh_size > 0) {
+    relocCount = static_cast<unsigned>(pRelocSection->secHead.sh_size / pRelocSection->secHead.sh_entsize);
+  }
+
+  return relocCount;
+}
+
+// =====================================================================================================================
+// Get the start pos of relocs used by PS in the reloc section
+//
+// @param pRelocSection : Reloc section
+// @param psIsaOffset : Ps ISA offset in the elf bin
+template <class Elf> size_t ElfWriter<Elf>::getRelocPsStartPos(const SectionBuffer *pRelocSection, size_t psIsaOffset) {
+  size_t psStartPos = 0;
+
+  size_t relocCount = getRelocsCount(pRelocSection);
+  if (relocCount > 0) {
```

Comment:
Can we flip this if condition and return early instead?
`if (relocCount == 0) return 0`;